File: game/random_player.py
import random as r
from core.action import *
from core.vector2 import Vector2
from player import Player


class RandomPlayer(Player):

    def take_action(self, opponent, board_state):
        if r.randint(0, 3) > 0:
            valid_moves = list(self.get_valid_move_positions(board_state))
            new_pos = valid_moves[r.randint(0, len(valid_moves) - 1)]
            return create_move(new_pos)
        else:
            wall_pos = Vector2(r.randint(0, 7), r.randint(0, 7))
            wall_orientation = 1 if r.randint(0, 1) == 0 else 2
            return create_block(wall_pos, wall_orientation)

File: game/src/random_player.cc
#include "random_player.h"
#include "validation.h"
#include <random>

RandomPlayer::RandomPlayer(double move_chance) : 
    Player(1),
    move_chance_(move_chance) {
}

Action RandomPlayer::TakeAction(const BoardState& board_state) {
    while (true) {
        if (static_cast<double>(rand()) / static_cast<double>(RAND_MAX) < move_chance_ 
        || board_state.GetPlayerWallCount(index_) == 0) {
            auto valid_moves = GetValidMoves(board_state);
            auto action = Action(valid_moves[rand() % valid_moves.size()]);
            if (ValidateAction(board_state, index_, action)) {
                return action;
            }
        }
        else {
            auto wall_pos = Vectori(rand() % 8, rand() % 8);
            auto wall_orientation = rand() % 2 + 1;
            auto action = Action(wall_pos, static_cast<WallOrientation>(wall_orientation));
            if (ValidateAction(board_state, index_, action)) {
                return action;
            }
        }
    }
}
